app/viewmodels/CashAccountV2ViewModel.scala

```diff
@@ -45,36 +47,77 @@ object CashAccountV2ViewModel {
 
+    val hasMaxTransactionsExceeded: Boolean = cashTrans.maxTransactionsExceeded.getOrElse(false)
+
     val dailyStatementsComponent: HtmlFormat.Appendable =
       new daily_statements_v2(emptyGovUkTableComponent).apply(CashAccountDailyStatementsViewModel(cashTrans))
 
     val cashAccountBalance: HtmlFormat.Appendable =
       new cash_account_balance(emptyH1Component, emptyH2InnerComponent, emptyPComponent)
-        .apply(model = CashAccountViewModel(eori, account))
+        .apply(model = CashAccountViewModel(eori, account), showLastTransactionsHeading = !hasMaxTransactionsExceeded)
 
     val requestTransactionsHeading: HtmlFormat.Appendable =
       h2Component(
         msgKey = "cf.cash-account.transactions.request-transactions.heading",
         id = Some("request-transactions-heading"))
 
+    val tooManyTransactionsHeading: Option[HtmlFormat.Appendable] = {
+      if (hasMaxTransactionsExceeded) {
+        Some(h2Component(
+          msgKey = "cf.cash-account.transactions.transactions-for-last-six-months.heading",
+          id = Some("last-six-month-transactions-heading")))
+      } else {
+        None
+      }
+    }
+
+    val tooManyTransactionsStatement: Option[HtmlFormat.Appendable] = {
+      if (hasMaxTransactionsExceeded) {
+        Some(pComponent(
+          id = Some("exceeded-threshold-statement"),
+          messageKey = "cf.cash-account.transactions.too-many-transactions.hint01",
+          classes = "govuk-body govuk-!-margin-bottom-0 govuk-!-margin-top-7"))
+      } else {
+        None
+      }
+    }
+
     CashAccountV2ViewModel(
       pageTitle = msgs("cf.cash-account.detail.title"),
       backLink = config.customsFinancialsFrontendHomepage,
       cashAccountBalance = cashAccountBalance,
       dailyStatements = dailyStatementsComponent,
       requestTransactionsHeading = requestTransactionsHeading,
-      downloadCSVFileLinkUrl = downloadCSVFileLinkUrl,
+      downloadCSVFileLinkUrl = downloadCSVFileLinkUrl(hasMaxTransactionsExceeded),
+      hasMaxTransactionsExceeded = hasMaxTransactionsExceeded,
+      tooManyTransactionsHeading = tooManyTransactionsHeading,
+      tooManyTransactionsStatement = tooManyTransactionsStatement,
       helpAndSupportGuidance = helpAndSupport)
   }
 
-  private def downloadCSVFileLinkUrl(implicit msgs: Messages): HtmlFormat.Appendable = {
-    linkComponent(
-      LinkComponentValues(
-        pId = Some("download-scv-file"),
-        linkMessageKey = "cf.cash-account.transactions.request-transactions.download-csv.url",
-        location = controllers.routes.RequestTransactionsController.onPageLoad().url,
-        postLinkMessageKey = Some("cf.cash-account.transactions.request-transactions.download-csv.post-message"),
-        enableLineBreakBeforePostMessage = true))
+  private def downloadCSVFileLinkUrl(hasMaxTransactionsExceeded: Boolean
+                                    )(implicit msgs: Messages): HtmlFormat.Appendable = {
+
+    if (hasMaxTransactionsExceeded) {
```

Comment:
Please do not keep the entire component inside if and else, just make the required field conditional and then use them

Reply:
discussed. Resolved